lightning/src/util/enforcing_trait_impls.rs

```diff
@@ -90,6 +107,20 @@
 	fn sign_holder_commitment<T: secp256k1::Signing + secp256k1::Verification>(&self, holder_commitment_tx: &HolderCommitmentTransaction, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()> {
 		// TODO: enforce the ChannelKeys contract - error if this commitment was already revoked
 		// TODO: need the commitment number
+		let revoked = self.revoked_commitment.lock().unwrap();
+		let keys = &holder_commitment_tx.keys;
+		if keys.per_commitment_point != self.inner.get_per_commitment_point(*revoked - 1, secp_ctx) {
```

Comment:
Wouldn't it be more legible to just create two variables for commitment points -1 and -2, and then compare that the keys match neither in one if statement? The first nested if has no else.

Reply:
The reason for the order here is the performance of the `get_per_commitment_point`.  The most common case is that we are signing the tx after the last revoked - `*revoked - 1`, so computing and checking that first is most performant for this happy path.  If that's not the case, signing the one after that (`*revoked - 2`) is the only remaining valid choice, so we check that.  If that's also not the case, we try to see if they made a logic mistake and are signing the last revoked so we can give an informative error message.  Lastly, if none of these match, we throw up our hands, since we don't know what's being signed.

Hmm... after writing that I realized that `*revoked - 2` is the common happy path, will swap those two.